lib/utils/fncache.go

```diff
@@ -170,6 +170,24 @@ func (c *FnCache) Shutdown(ctx context.Context) {
 	}
 }
 
+// Remove purges a specific item in the cache.
+func (c *FnCache) Remove(key any) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	delete(c.entries, key)
+}
+
+// Set set an item in the cache.
+func (c *FnCache) Set(key any, v any, ttl time.Duration) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	c.entries[key] = &fnCacheEntry{
+		v:   v,
+		t:   c.cfg.Clock.Now(),
+		ttl: ttl,
+	}
+}
```

Comment:
We definitely want test coverage for the new methods.  Also, I believe that the entry being added here needs to have its `loaded` channel be a valid closed channel, not nil, in order for subsequent reads & ttl cleanups to work correctly.